gui/syncthing/core/eventService.js

```diff
@@ -18,14 +28,42 @@ angular.module('syncthing.core')
                 return;
             }
             $rootScope.$broadcast(self.ONLINE);
-
+            
+            
             if (lastID > 0) {   // not emit events from first response
+                var greenCounter = 0;
+                var redCounter = 0;
+                
```

Comment:
instead of counting each color in a separate variable you should use a single variable that holds the icon state, e.g. 'error'